### PR TITLE
Decompose webSessionSetup into reusable parts (Phase 3)

### PR DESCRIPTION
**Pull request by kraenhansen**
## Summary

- Extract `setupWebRTCSession()` into common `platform/VoiceSessionSetup.ts` — platform-agnostic WebRTC session setup that extracts input/output from `WebRTCConnection`
- Replace web `setupInputOutput` (handled both connection types) with `setupWebSocketIO` (WebSocket-only), delegating WebRTC to the common function
- Update `internal.ts` exports: remove `webSessionSetup` (which imported web-only code), add `setupWebRTCSession` + `createConnection`
- Update React Native package to call `createConnection()` + `setupWebRTCSession()` directly instead of `webSessionSetup()`

This fixes all TS6307 errors where `internal.ts` was dragging `platform/web/` files into the common tsconfig build. React Native no longer transitively imports web-only `MediaDeviceInput`/`MediaDeviceOutput`.

Stacked on #784 (Phase 2).

### `webSessionSetup` removed from `@elevenlabs/client/internal`

**Before:**
```ts
import { webSessionSetup } from "@elevenlabs/client/internal";

const result = await webSessionSetup(options);
```

**After:** Use `createConnection` + `setupWebRTCSession` for WebRTC, or the registered platform strategy for full setup:
```ts
import { createConnection, setupWebRTCSession } from "@elevenlabs/client/internal";

const connection = await createConnection(options);
const result = setupWebRTCSession(connection);
```

**Why:** `webSessionSetup` imported web-only code (MediaDeviceInput/Output, AudioContext), which pulled DOM dependencies into any consumer — including React Native. Splitting into `createConnection` (platform-agnostic) and `setupWebRTCSession` (also platform-agnostic) lets non-web platforms set up voice sessions without importing web APIs.

---

### `setupInputOutput` removed from `platform/web/VoiceSessionSetup.ts` exports

**Before:**
```ts
import { setupInputOutput } from "@elevenlabs/client/platform/web/VoiceSessionSetup";
```

**After:** This function is no longer exported. WebRTC setup is handled by `setupWebRTCSession` from `platform/VoiceSessionSetup.ts`. WebSocket I/O setup is internal to the web strategy.

**Why:** `setupInputOutput` mixed WebRTC and WebSocket concerns in a single function, making it impossible to use either path independently. WebRTC setup is now platform-agnostic, and WebSocket setup is an internal detail of the web platform strategy.

## Changes

- **`packages/client/src/internal.ts`** — Remove `webSessionSetup` export, add `setupWebRTCSession` and `createConnection` exports
- **`packages/client/src/platform/VoiceSessionSetup.ts`** — Add `setupWebRTCSession()` function for platform-agnostic WebRTC session setup
- **`packages/client/src/platform/web/VoiceSessionSetup.ts`** — Replace exported `setupInputOutput` with private `setupWebSocketIO`; `webSessionSetup` now delegates WebRTC to `setupWebRTCSession`
- **`packages/react-native/src/index.react-native.ts`** — Use `createConnection()` + `setupWebRTCSession()` instead of `webSessionSetup()`

## Test plan

- [x] TS6307 errors eliminated (verified: 0 occurrences)
- [x] React Native package passes `tsc --noEmit`
- [x] 65/65 unit tests passing
- [ ] Manual test: web voice conversation (WebRTC path)
- [ ] Manual test: web voice conversation (WebSocket path)
- [ ] Manual test: React Native voice conversation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Medium risk due to a breaking change in `@elevenlabs/client/internal` exports and refactoring of session setup branching across web and React Native, which could affect voice initialization paths if edge cases were missed.
> 
> **Overview**
> **WebRTC session setup is extracted into a reusable, platform-agnostic helper.** `platform/VoiceSessionSetup.ts` now exposes `setupWebRTCSession()` (with type-checking and clearer errors) and adds unit tests for expected/invalid inputs.
> 
> **Web and React Native setup paths are refactored to use the new split.** Web `webSessionSetup` now only builds MediaDevice I/O for `WebSocketConnection` and delegates WebRTC to `setupWebRTCSession`, while React Native stops importing the web strategy, explicitly creates a connection via `createConnection`, uses `setupWebRTCSession`, and throws early if WebSocket/signedUrl options are provided. `internal.ts` updates exports to drop `webSessionSetup` and instead export `setupWebRTCSession` plus `createConnection`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b4f4512c6dd7253893af2f92f77b3e3e3ee603e4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/client/src/platform/web/VoiceSessionSetup.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import type { Options } from "../../BaseConversation.js";
 import type { BaseConnection } from "../../utils/BaseConnection.js";
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply packages/client/src/platform/web/VoiceSessionSetup.ts:2]: Fixed in Phase 4 (b4f4512 → 88eaf14): the web setup file was rewritten and `BaseConnection` is no longer imported. `WebRTCConnection` is still used for the `instanceof` check in the WebRTC branch.

File: packages/client/src/platform/VoiceSessionSetup.ts [hunk 2]
```diff
@@ -30,3 +31,26 @@ export let setupStrategy: VoiceSessionSetupStrategy | undefined;
...
+ * Platform-agnostic: extracts the input/output controllers that
+ * WebRTCConnection already provides via LiveKit.
+ */
+export function setupWebRTCSession(
+  connection: BaseConnection
+): VoiceSessionSetupResult {
+  if (!(connection instanceof WebRTCConnection)) {
+    throw new Error(
+      "setupWebRTCSession requires a WebRTCConnection. " +
+        `Received: ${connection.constructor.name}`
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply packages/client/src/platform/VoiceSessionSetup.ts]: Fixed in b4f4512: changed to `connection?.constructor?.name ?? typeof connection` so null/undefined inputs produce `"Received: object"` or `"Received: undefined"` instead of throwing a TypeError. Added tests for both edge cases.

File: packages/react-native/src/index.react-native.ts [hunk 3]
```diff
@@ -36,7 +37,8 @@ async function reactNativeSessionSetup(
   });
   await AudioSession.startAudioSession();
 
-  const result = attachNativeVolume(await webSessionSetup(options));
+  const connection = await createConnection(options);
+  const result = attachNativeVolume(setupWebRTCSession(connection));
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply packages/react-native/src/index.react-native.ts:54]: Fixed in b4f4512: added an early guard that checks for `connectionType: "websocket"` or `signedUrl` and throws a clear RN-specific error: *"WebSocket connections are not supported on React Native."* Also updated the doc comment to document the limitation.

File: packages/client/src/platform/web/VoiceSessionSetup.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import type { Options } from "../../BaseConversation.js";
...
 import {
   setSetupStrategy,
+  setupWebRTCSession,
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply packages/client/src/platform/web/VoiceSessionSetup.ts:5]: Fixed in Phase 4 (88eaf14): the web setup file was rewritten — `WebRTCConnection` is used again (for the `instanceof` check in the WebRTC branch) and `BaseConnection` is no longer imported.
